python/ray/workers/setup_worker.py

```diff
@@ -119,5 +120,15 @@ def start_worker_in_container(container_option, args, remaining_args):
     else:
         remaining_args.append("--serialized-runtime-env")
         remaining_args.append(args.serialized_runtime_env or "{}")
-        setup = import_attr(args.worker_setup_hook)
-        setup(remaining_args)
+        remaining_args.append("--worker-shim-pid={}".format(os.getpid()))
+
+        def start_worker():
+            setup = import_attr(args.worker_setup_hook)
+            setup(remaining_args)
+
+        if args.allocated_instances_serialized_json != "{}":
+            import setup_cgroup
+            setup_cgroup.start_worker_in_cgroup(
```

Comment:
Check if cgroup is available here

Reply:
I added the check in`start_worker_in_cgroup`, PTAL